File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int len = p.length();
        for(int i = 0; i < t.length() - len + 1; i++) {
            if(stol(p) >= stol(t.substring(i, i + len)))
                answer++;
        }
        return answer;
    }
    static long stol(String s) {
        return Long.parseLong(s);
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    var answer = 0;
    var len = p.length;
    var num = Number(p);
    for(let i = 0; i <= t.length - len; i++) {
        if(Number(t.slice(i, i + len)) <= num) answer++;
    }
    return answer;
}